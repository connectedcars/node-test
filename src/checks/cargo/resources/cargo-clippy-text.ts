import { CargoMessage } from '../cargo-types'

export const cargoClippyFailedOutput: CargoMessage[] = [
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered:
        'error: this comparison involving the minimum or maximum element for this type contains a case that is always true or always false\n   --> can/src/can_session.rs:391:12\n    |\n391 |         if 100 > i32::MAX {}\n    |            ^^^^^^^^^^^^^^\n    |\n    = note: `-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`\n    = help: because `i32::MAX` is the maximum value for this type, this comparison is always false\n    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons\n\n',
      children: [
        {
          children: [],
          code: null,
          level: 'note',
          message: '`-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message: 'because `i32::MAX` is the maximum value for this type, this comparison is always false',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message:
            'for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'clippy::absurd_extreme_comparisons',
        explanation: null
      },
      level: 'error',
      message:
        'this comparison involving the minimum or maximum element for this type contains a case that is always true or always false',
      spans: [
        {
          byte_end: 13338,
          byte_start: 13324,
          column_end: 26,
          column_start: 12,
          expansion: null,
          file_name: 'can/src/can_session.rs',
          is_primary: true,
          label: null,
          line_end: 391,
          line_start: 391,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 26,
              highlight_start: 12,
              text: '        if 100 > i32::MAX {}'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered: 'error: aborting due to previous error\n\n',
      children: [],
      code: null,
      level: 'error',
      message: 'aborting due to previous error',
      spans: []
    }
  }
]
