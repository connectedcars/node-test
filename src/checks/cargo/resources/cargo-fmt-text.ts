import { CargoFmtFile } from '../cargo-types'

export const cargoFmtFailedOutput: CargoFmtFile[] = [
  {
    name: '/checks-test/src/foo/bar.rs',
    mismatches: [
      {
        original_begin_line: 1,
        original_end_line: 7,
        expected_begin_line: 1,
        expected_end_line: 2,
        original: '\n\npub fn bar(\n\n)\n{\n    println!  (  "bar"  )  ;',
        expected: 'pub fn bar() {\n    println!("bar");'
      }
    ]
  },
  {
    name: '/checks-test/src/foo/mod.rs',
    mismatches: [
      {
        original_begin_line: 1,
        original_end_line: 5,
        expected_begin_line: 1,
        expected_end_line: 1,
        original: '\n\nmod bar\n;\n',
        expected: 'mod bar;'
      }
    ]
  },
  {
    name: '/checks-test/src/main.rs',
    mismatches: [
      {
        original_begin_line: 9,
        original_end_line: 9,
        expected_begin_line: 9,
        expected_end_line: 9,
        original: '',
        expected: 'mod foo;'
      },
      {
        original_begin_line: 10,
        original_end_line: 10,
        expected_begin_line: 11,
        expected_end_line: 12,
        original: ' mod foo  ;',
        expected: 'fn main() {\n    let foo = foo();'
      },
      {
        original_begin_line: 12,
        original_end_line: 16,
        expected_begin_line: 14,
        expected_end_line: 14,
        original: '\nfn main()\n{\n      let foo = foo();\n',
        expected: ''
      },
      {
        original_begin_line: 19,
        original_end_line: 19,
        expected_begin_line: 16,
        expected_end_line: 16,
        original: '',
        expected: ''
      },
      {
        original_begin_line: 27,
        original_end_line: 28,
        expected_begin_line: 23,
        expected_end_line: 23,
        original: '\n',
        expected: ''
      },
      {
        original_begin_line: 35,
        original_end_line: 38,
        expected_begin_line: 29,
        expected_end_line: 32,
        original: '\n\n    if { let b = false; b } {}\n',
        expected: '    if {\n        let b = false;\n        b\n    } {}'
      },
      {
        original_begin_line: 56,
        original_end_line: 56,
        expected_begin_line: 50,
        expected_end_line: 50,
        original: '    assert!(false);',
        expected: '        assert!(false);'
      },
      {
        original_begin_line: 71,
        original_end_line: 72,
        expected_begin_line: 65,
        expected_end_line: 65,
        original: '\n',
        expected: ''
      },
      {
        original_begin_line: 74,
        original_end_line: 75,
        expected_begin_line: 66,
        expected_end_line: 66,
        original: '\n',
        expected: ''
      }
    ]
  }
]
