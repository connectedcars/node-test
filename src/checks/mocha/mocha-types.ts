export interface MochaData {
  stats: Stats
  tests: MochaTest[]
  failures: MochaTest[]
  passes?: MochaTest[]
  pending?: MochaTest[]
}

interface MochaTest {
  title?: string
  fullTitle?: string
  duration: number
  currentRetry: number
  err: MochaError
}

interface MochaError {
  message?: string
  stack?: string
  errorMode?: string
  parent?: MochaParent
  name?: string
  hasSerializedErrorMessage?: boolean
  fullTitle?: string
  title?: string
}

interface MochaParent {
  errorMode: string
  parent: MochaParent | null
  name: string
  label?: string
}

interface Stats {
  suites: number
  tests: number
  passes: number
  pending: number
  failures: number
  start?: string
  end?: string
  duration?: number
}
