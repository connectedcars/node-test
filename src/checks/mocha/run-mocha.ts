import { runJsonCommand } from '../checks-common'

export async function runMocha(extraArgs: string[] = []): Promise<MochaData> {
  const [_, json] = await runJsonCommand<MochaData>('mocha', [...extraArgs, 'src/**/*.test.js', '--reporter=json'], {
    env: { ...process.env, TZ: 'UTC' }
  })
  return json
}
