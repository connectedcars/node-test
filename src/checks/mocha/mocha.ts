/* eslint-disable no-console */
import { CheckAnnotation, CheckConversionError, CheckRunCompleted } from '../checks-common'
import { MochaData } from './mocha-types'

export interface MochaInput {
  sha: string
  data: MochaData
}

export function mochaCheck({ data, sha }: MochaInput): CheckRunCompleted {
  try {
    const annotations: CheckAnnotation[] = []

    // This happens when a test logs something to stdout while running
    if (!data || !data.failures) {
      return {
        name: 'mocha',
        head_sha: sha,
        conclusion: 'neutral',
        status: 'completed',
        completed_at: new Date().toISOString(),
        output: {
          title: 'No tests found',
          summary: 'No tests found',
          annotations: []
        }
      }
    }

    // Run through each failed test to create annotations
    for (const test of data.failures) {
      if (!test.err?.stack) {
        continue
      }
      // Find file path from stack
      const matches = test.err.stack.match(/Context\.(?:[^\s]+) \((.+?)\)/s)
      if (!matches) {
        console.log('fail', test.err.stack)
        continue
      }
      const [filePath, line, column] = matches[1].split(':')
      const lineNumber = parseInt(line)

      if (test.err.message && test.err.message.length > 4 * 1024) {
        test.err.message = test.err.message.substring(0, 4 * 1024)
      }
      if (test.err.stack && test.err.stack.length > 4 * 1024) {
        test.err.stack = test.err.stack.substring(0, 4 * 1024)
      }

      // Generate an annotation
      annotations.push({
        path: filePath,
        start_line: lineNumber,
        end_line: lineNumber,
        annotation_level: 'failure',
        message: `${line}:${column}`.padEnd(10) + `${test.title} ${test.err.message}`,
        title: `${filePath}#L${line}`,
        raw_details: JSON.stringify(test, null, '    ')
      })
    }
    // Generate a summary of the problems
    const failing = data.stats.failures
    const passing = data.stats.passes
    const pending = data.stats.pending
    let summary = `Found **${failing}** failed ${failing === 1 ? 'test' : 'tests'}`
    const details = [`**${passing}** passing`]
    if (pending > 0) {
      details.push(`**${pending}** pending`)
    }
    summary += ` (${details.join(', ')})`
    return {
      name: 'mocha',
      head_sha: sha,
      conclusion: failing > 0 ? 'failure' : 'success',
      status: 'completed',
      completed_at: new Date().toISOString(),
      output: {
        title: 'mocha',
        summary,
        annotations
      }
    }
  } catch (e) {
    throw new CheckConversionError('mocha', { data, sha }, e as Error)
  }
}
