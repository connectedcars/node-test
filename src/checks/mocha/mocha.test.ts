import { mochaCheck } from './mocha'
import { mochaFailedOutput, mochaSuccessfulOutput } from './resources/mocha-help-text'

describe('checks/mocha', () => {
  it('processes passing mocha output to checks structure', () => {
    const data = JSON.parse(mochaSuccessfulOutput)

    const output = mochaCheck({
      data,
      sha: 'c61a4ae014360e064eb2a9f76c8a6a55d05e5b88'
    })
    const expected = {
      name: 'mocha',
      head_sha: 'c61a4ae014360e064eb2a9f76c8a6a55d05e5b88',
      conclusion: 'success',
      status: 'completed',
      completed_at: expect.stringMatching(/^\d{4}/),
      output: {
        title: 'mocha',
        summary: 'Found **0** failed tests (**2787** passing)',
        annotations: []
      }
    }
    expect(output).toStrictEqual(expected)
  })

  it('processes failing mocha output to checks structure', () => {
    const data = mochaFailedOutput

    const output = mochaCheck({
      data,
      sha: 'c61a4ae014360e064eb2a9f76c8a6a55d05e5b88'
    })
    expect(output).toMatchSnapshot({
      completed_at: expect.stringMatching(/^\d{4}/)
    })
  })
})
