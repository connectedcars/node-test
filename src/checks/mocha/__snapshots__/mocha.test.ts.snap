// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks/mocha processes failing mocha output to checks structure 1`] = `
Object {
  "completed_at": StringMatching /\\^\\\\d\\{4\\}/,
  "conclusion": "failure",
  "head_sha": "c61a4ae014360e064eb2a9f76c8a6a55d05e5b88",
  "name": "mocha",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "failure",
        "blob_href": "http://github.com/connectedcars/cloudbuilder-wrapper/blob/c61a4ae014360e064eb2a9f76c8a6a55d05e5b88/src/data/db/vehicles.get-optimizations.test.js",
        "end_line": 825,
        "message": "825:7     gets optimizations by allowed workshop ids 
[31m[1mexpected[22m[39m [36m'select \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId!\` is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?'[39m
[31m[1mto be[22m[39m [36m'select \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId\` is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?'[39m

[31mselect \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId[39m[41m[30m!\`[39m[49m[31m is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?[39m
[32mselect \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId[39m[42m[30m\`[39m[49m[32m is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?[39m
",
        "path": "src/data/db/vehicles.get-optimizations.test.js",
        "raw_details": "{
    \\"title\\": \\"gets optimizations by allowed workshop ids\\",
    \\"fullTitle\\": \\"db/vehicles getOptimizations gets optimizations by allowed workshop ids\\",
    \\"duration\\": 12,
    \\"currentRetry\\": 0,
    \\"err\\": {
        \\"errorMode\\": \\"default\\",
        \\"stack\\": \\"UnexpectedError: \\\\n\\\\u001b[31m\\\\u001b[1mexpected\\\\u001b[22m\\\\u001b[39m \\\\u001b[36m'select \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId!\` is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?'\\\\u001b[39m\\\\n\\\\u001b[31m\\\\u001b[1mto be\\\\u001b[22m\\\\u001b[39m \\\\u001b[36m'select \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId\` is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?'\\\\u001b[39m\\\\n\\\\n\\\\u001b[31mselect \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId\\\\u001b[39m\\\\u001b[41m\\\\u001b[30m!\`\\\\u001b[39m\\\\u001b[49m\\\\u001b[31m is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?\\\\u001b[39m\\\\n\\\\u001b[32mselect \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId\\\\u001b[39m\\\\u001b[42m\\\\u001b[30m\`\\\\u001b[39m\\\\u001b[49m\\\\u001b[32m is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?\\\\u001b[39m\\\\n\\\\n    at Context.it (src/data/db/vehicles.get-optimizations.test.js:825:7)\\\\n    set UNEXPECTED_FULL_TRACE=true to see the full stack trace\\",
        \\"parent\\": {
            \\"errorMode\\": \\"default\\",
            \\"parent\\": {
                \\"errorMode\\": \\"default\\",
                \\"parent\\": null,
                \\"name\\": \\"UnexpectedError\\",
                \\"label\\": \\"should equal\\"
            },
            \\"name\\": \\"UnexpectedError\\"
        },
        \\"name\\": \\"UnexpectedError\\",
        \\"message\\": \\"\\\\n\\\\u001b[31m\\\\u001b[1mexpected\\\\u001b[22m\\\\u001b[39m \\\\u001b[36m'select \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId!\` is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?'\\\\u001b[39m\\\\n\\\\u001b[31m\\\\u001b[1mto be\\\\u001b[22m\\\\u001b[39m \\\\u001b[36m'select \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId\` is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?'\\\\u001b[39m\\\\n\\\\n\\\\u001b[31mselect \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId\\\\u001b[39m\\\\u001b[41m\\\\u001b[30m!\`\\\\u001b[39m\\\\u001b[49m\\\\u001b[31m is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?\\\\u001b[39m\\\\n\\\\u001b[32mselect \`cars\`.\`id\`, \`cars\`.\`obdId\`, \`cars\`.\`vin\`, \`cars\`.\`name\`, \`cars\`.\`label\`, \`cars\`.\`imageFilename\`, \`cars\`.\`activated\`, \`cars\`.\`deviceActivated\`, \`cars\`.\`workshopId\`, null as \`closedAt\`, null as \`assignedUserId\`, \`cars\`.\`createdAt\`, \`cars\`.\`updatedAt\`, max(\`LicensePlates\`.\`licensePlate\`) as \`licensePlate\` from \`cars\` left join \`LicensePlates\` on \`LicensePlates\`.\`carId\` = \`cars\`.\`id\` where (\`cars\`.\`workshopId\` in (?, ?) or \`cars\`.\`workshopId\\\\u001b[39m\\\\u001b[42m\\\\u001b[30m\`\\\\u001b[39m\\\\u001b[49m\\\\u001b[32m is null) and \`cars\`.\`archived\` = ? group by \`cars\`.\`id\` limit ?\\\\u001b[39m\\\\n\\",
        \\"hasSerializedErrorMessage\\": true
    }
}",
        "start_line": 825,
        "title": "src/data/db/vehicles.get-optimizations.test.js#L825",
      },
      Object {
        "annotation_level": "failure",
        "blob_href": "http://github.com/connectedcars/cloudbuilder-wrapper/blob/c61a4ae014360e064eb2a9f76c8a6a55d05e5b88/src/utils/cc-api/connected-cars-api.test.js",
        "end_line": 111,
        "message": "111:7     undefined undefined",
        "path": "src/utils/cc-api/connected-cars-api.test.js",
        "raw_details": "{
    \\"currentRetry\\": 0,
    \\"duration\\": 8,
    \\"err\\": {
        \\"hasSerializedErrorMessage\\": true,
        \\"errorMode\\": \\"default\\",
        \\"stack\\": \\"UnexpectedError: expected 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjEifQ.eyJhdWQiOiJjb25uZWN0ZWRjYXJzL2FwcCIsImlzcyI6Imh0dHBzOi8vYXV0aC1hcGkiLCJpYXQiOjE1ODIyNDMyMDAsImV4cCI6MTU4MjI0NjgwMH0.gOdbG9xSWKvBbUx9aiO_6QB_iZh4aM2Hw2YTp4XAdqGAq40k43ytaFq26QWe3oOst7l3FyEnzEDpahwWFOoL2S_Ep-GBRhZMPq9MU00gWIPPojVga89qi4-lgEmZtlo2_AL3_wmUMc35VlWfUVmNV0iL1UVyzuLIuMpbS6OBnLQ' to equal 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjEifQ.eyJhdWQiOiJjb25uZWN0ZWRjYXJzL2FwcCIsImlzcyI6Imh0dHBzOi8vYXV0aC1hcGkiLCJpYXQiOjE1ODIyMzk2MDAsImV4cCI6MTU4MjI0MzIwMH0.ba977JlCDuIA4L1RIutbvDdqtJ4__Nh8-TF33xZtY0bEf0ggGis_55IGxms_J1UraTZ8IavfebT7m18Op6Fl8ML8LU0esDadhl83VmQA6n77MvtOiiu2fR4VwMPX4Ih6avRKEfIa4DxBc4rhEv90ZYmyhpUfDFemreta92K1rpw' -eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjEifQ.eyJhdWQiOiJjb25uZWN0ZWRjYXJzL2FwcCIsImlzcyI6Imh0dHBzOi8vYXV0aC1hcGkiLCJpYXQiOjE1ODIyNDMyMDAsImV4cCI6MTU4MjI0NjgwMH0.gOdbG9xSWKvBbUx9aiO_6QB_iZh4aM2Hw2YTp4XAdqGAq40k43ytaFq26QWe3oOst7l3FyEnzEDpahwWFOoL2S_Ep-GBRhZMPq9MU00gWIPPojVga89qi4-lgEmZtlo2_AL3_wmUMc35VlWfUVmNV0iL1UVyzuLIuMpbS6OBnLQ +eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjEifQ.eyJhdWQiOiJjb25uZWN0ZWRjYXJzL2FwcCIsImlzcyI6Imh0dHBzOi8vYXV0aC1hcGkiLCJpYXQiOjE1ODIyMzk2MDAsImV4cCI6MTU4MjI0MzIwMH0.ba977JlCDuIA4L1RIutbvDdqtJ4__Nh8-TF33xZtY0bEf0ggGis_55IGxms_J1UraTZ8IavfebT7m18Op6Fl8ML8LU0esDadhl83VmQA6n77MvtOiiu2fR4VwMPX4Ih6avRKEfIa4DxBc4rhEv90ZYmyhpUfDFemreta92K1rpw at Context.<anonymous> (src/utils/cc-api/connected-cars-api.test.js:111:7) set UNEXPECTED_FULL_TRACE=true to see the full stack trace\\",
        \\"fullTitle\\": \\"utils/cc-api/connectedcars-api.js getAccessToken gets a new token if none exists and then reuses it\\",
        \\"title\\": \\"gets a new token if none exists and then reuses it\\"
    }
}",
        "start_line": 111,
        "title": "src/utils/cc-api/connected-cars-api.test.js#L111",
      },
    ],
    "summary": "Found **4** failed tests (**3068** passing, **1** pending)",
    "title": "mocha",
  },
  "status": "completed",
}
`;
